workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/WorkflowInterceptor.kt

```diff
@@ -129,6 +142,29 @@ public interface WorkflowInterceptor {
     session: WorkflowSession
   ): Snapshot? = proceed(state)
 
+  /**
+   * Called when a the runtime loop has "iterated." Typically this the application of 1 action
+   * and 1 render pass.
+   *
+   * If there is no state change with the `RENDER_ONLY_WHEN_STATE_CHANGES` optimization,
+   * it might not be a render pass at all, in which case [RenderPassSkipped] is the outcome.
+   * In the case of the `CONFLATE_STALE_RENDERINGS` optimization it could be multiple render
+   * passes.
+   * If there is at least one render pass, then [RenderPassesComplete] is passed as the outcome,
+   * which includes the actual [RenderingAndSnapshot] returned from the runtime.
+   *
+   * @param outcome The [RuntimeLoopOutcome] of the runtime iteration.
+   */
+  public fun onRuntimeLoopIteration(
+    outcome: RuntimeLoopOutcome
+  ): Unit = Unit
+
+  public sealed interface RuntimeLoopOutcome
+  public data object RenderPassSkipped : RuntimeLoopOutcome
+  public data class RenderPassesComplete<R>(
+    val renderingAndSnapshot: RenderingAndSnapshot<R>
+  ) : RuntimeLoopOutcome
```

Comment:
We've avoided `data class` and `data object` in public types.

Reply:
Thanks for catching that. Will fix.